zkevm-circuits/src/evm_circuit/witness.rs

```diff
@@ -557,11 +558,46 @@ impl Rw {
     }
 }
 
+impl From<&ExecError> for ExecutionState {
+    fn from(error: &ExecError) -> Self {
+        match error {
+            ExecError::ExecutionReverted => ExecutionState::ErrorReverted,
+            ExecError::InvalidOpcode => ExecutionState::ErrorInvalidOpcode,
+            ExecError::StackOverflow => ExecutionState::ErrorStackOverflow,
+            ExecError::StackUnderflow => ExecutionState::ErrorStackUnderflow,
+            ExecError::WriteProtection => ExecutionState::ErrorWriteProtection,
+            ExecError::Depth => ExecutionState::ErrorDepth,
+            ExecError::InsufficientBalance => ExecutionState::ErrorInsufficientBalance,
+            ExecError::ContractAddressCollision => ExecutionState::ErrorContractAddressCollision,
+            ExecError::InvalidCode => ExecutionState::ErrorInvalidCreationCode,
```

Comment:
in fact not sure whether `ExecError::InvalidCode` is corresonding to `ExecutionState::ErrorInvalidCreationCode`.

But would like to fix if this line wrong

Reply:
I think that is correct.
- `InvalidOpcode`: happens when the execution encounters an opcode that is not valid
- `InvalidCode`: happens when a creation call returns a code that is invalid (starts with `0xef`)

Maybe we can rename `ExecError::InvalidCode` to `ExecError::InvalidCreationCode` to be consistent.

Also we can rename `ExecError::ExecutionReverted` to `ExecError::Reverted` for consistency.